src/LogicTasks/Semantics/Fill.hs

```diff
@@ -152,26 +160,30 @@ partialGrade FillInst{..} sol = do
       solLen = length boolSol
       missingLen = length missing
 
-completeGrade :: OutputCapable m => FillInst -> [TruthValue] -> LangM m
-completeGrade FillInst{..} sol = do
-  preventWithHint (notNull diff)
-    (translate $ do
-      german "Lösung ist korrekt?"
-      english "Solution is correct?"
-    )
-    (do
-      translate $ do
-        german $ "Die Lösung beinhaltet " ++ displayMistake ++ " Fehler."
-        english $ "Your solution contains " ++ displayMistake ++ " mistakes."
-      when showSolution $ example (show missingValues) $ do
-        english "The solution for this task is:"
-        german "Die Lösung für die Aufgabe ist:"
-      pure ()
-    )
+completeGrade :: (OutputCapable m, Alternative m, Monad m) => FillInst -> [TruthValue] -> Rated m
+completeGrade FillInst{..} sol = reRefuse
+  (extendedMultipleChoice
+    (MinimumThreshold (1 % 2))
+    (Punishment 0)
+    (TargetedCorrect (length solution))
+    DefiniteArticle
+    what
+    solutionDisplay
+    solution
+    submission)
+  $ translate $ do
+    german $ "Die Lösung beinhaltet " ++ displayMistake ++ " Fehler."
+    english $ "Your solution contains " ++ displayMistake ++ " mistakes."
```

Comment:
Won't this be printed now even when the student's submission was correct?

And when the student's submission was incorrect and a correct solution was shown due to `showSolution`, it could be unclear here (depending on the order of output, I guess) what "Die Lösung" is meant as referred here.

Reply:
I updated the wording and the conditions for the hints. The hint won't be shown now if `showSolution` is set to `True`.